themes/mlibrary_new/custom.php

```diff
@@ -127,11 +101,21 @@ function mlibrary_new_display_popular_tags()
 
 function mlibrary_new_display_exhibit_card_info($exhibitPage)
 {  // to see if there is no image attached to the first page.
+  $page_card_info = [];
+  $page_description = '';
+  $page_title = get_view()->shortcodes(metadata($exhibitPage, 'title'));
+
+  // Each page must have at least one Block and an Image must be added.
+  if ((!empty($exhibitPage->getPageBlocks())) and (!empty($exhibitPage->getAllAttachments()))) {
```

Comment:
Negated `and` statements are weird here.
Also, calling `getPageBlocks()` and `getAllAttachments()` twice seems unnecessary.
I recommend applying DeMorgan's law to get rid of the double negations, and peel off the new outer negation by swapping clauses.
Then eliminate the else clause by having stored the `$block` and `$rawAttachment` earlier.

Reply:
"Then eliminate the else clause by having stored the $block and $rawAttachment earlier" I have to check first if the block exists before store it. I will change the structure.